tools/shared/src/main/scala/org/scalajs/core/tools/sourcemap/SourceMapWriter.scala

```diff
@@ -194,18 +239,25 @@ class SourceMapWriter(
    */
   private def writeBase64VLQ(value0: Int): Unit = {
     // Sign is encoded in the least significant bit
-    var value =
-      if (value0 < 0) ((-value0) << 1) + 1
-      else value0 << 1
+    val signExtended = value0 >> 31
+    val value = (((value0 ^ signExtended) - signExtended) << 1) | (signExtended & 1)
```

Comment:
This really really needs a good comment. I suggest:
```scala
/* The sign is encoded in the least significant bit, while the
 * absolute value is shifted one bit to the left.
 * So in theory the "definition" of `value` is:
 *   val value =
 *     if (value0 < 0) ((-value0) << 1) | 1
 *     else value0 << 1
 * The following code is a branchless version of that spec.
 * It is valid because:
 * - if value0 < 0:
 *   signExtended == value0 >> 31 == 0xffffffff
 *   value0 ^ signExtended == ~value0
 *   (value0 ^ signExtended) - signExtended == ~value0 - (-1) == -value0
 *   signExtended & 1 == 1
 *   So we get ((-value0) << 1) | 1 as required
 * - if n >= 0:
 *   signExtended == value0 >> 31 == 0
 *   value0 ^ signExtended == value0
 *   (value0 ^ signExtended) - signExtended == value0 - 0 == value0
 *   signExtended & 1 == 0
 *   So we get (value0 << 1) | 0 == value0 << 1 as required
 */
```